protocol-designer/src/load-file/migration/8_5_0.ts

```diff
@@ -87,7 +87,9 @@ const getClippedFlowRateForMoveLiquid = (args: {
   const tipLiquidSpecs = liquidClass?.byPipette
     .find(byPipette => byPipette.pipetteModel === pipetteName)
     ?.byTipType.find(byTipType => byTipType.tiprack === tiprack)
+  // TODO: this won't find the definition for a custom tiprack
   const tiprackDef = getAllDefinitions()[tiprack]
```

Comment:
for types to agree when passing this arg
```suggestion
  const tiprackDef = getAllDefinitions()[tiprack] ?? null
```

Reply:
Heh, so we're relying in a TypeScript quirk to make the code say what we want, even though we know the implementation is not quite correct:

- We want users to always specify the `tiprackDef` when calling `getMatchingTipLiquidSpecsFromSpec()`, so `tiprackDef` is defined as non-null there.
- And when we write `getAllDefinitions()[tiprack]`, TypeScript expects to always find the `tiprackDef` here, so TypeScript also treats this result here as non-null.

Basically, this code here and `getMatchingTipLiquidSpecsFromSpec()` are written to not crash if `tiprackDef` happens to be null, but morally, we're saying that `tiprackDef` must not be null.